sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.encryption;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosAsyncDatabase;
+import com.azure.cosmos.CosmosBridgeInternal;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosDatabase;
+import com.microsoft.data.encryption.cryptography.EncryptionKeyStoreProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * CosmosClient with encryption support.
+ */
+public class CosmosEncryptionClient {
+    private final static Logger LOGGER = LoggerFactory.getLogger(CosmosEncryptionAsyncClient.class);
+    private final CosmosEncryptionAsyncClient cosmosEncryptionAsyncClient;
+    private EncryptionKeyStoreProvider encryptionKeyStoreProvider;
+    private final CosmosAsyncClient cosmosAsyncClient;
+    private final CosmosClient cosmosClient;
+
+    CosmosEncryptionClient(CosmosClient cosmosClient, EncryptionKeyStoreProvider encryptionKeyStoreProvider) {
+        this.encryptionKeyStoreProvider = encryptionKeyStoreProvider;
+        this.cosmosClient = cosmosClient;
+        this.cosmosAsyncClient = CosmosBridgeInternal.getCosmosAsyncClient(cosmosClient);
+        this.cosmosEncryptionAsyncClient = new CosmosEncryptionAsyncClient(cosmosAsyncClient, encryptionKeyStoreProvider);
+    }
+
+    public EncryptionKeyStoreProvider getEncryptionKeyStoreProvider() {
```

Comment:
does `EncryptionKeyStoreProvider` also expose sync api? or does it have any APIs with Mono?

Reply:
it expose sync apis